spring-webmvc/src/main/java/org/springframework/web/servlet/mvc/method/annotation/HttpEntityMethodProcessor.java

```diff
@@ -237,18 +239,18 @@ else if (returnValue instanceof ProblemDetail detail) {
 		}
 
 		if (httpEntity instanceof ResponseEntity<?> responseEntity) {
-			int returnStatus = responseEntity.getStatusCode().value();
-			outputMessage.getServletResponse().setStatus(returnStatus);
-			if (returnStatus == 200) {
+			HttpStatusCode httpStatusCode = responseEntity.getStatusCode();
+			outputMessage.getServletResponse().setStatus(httpStatusCode.value());
+			if (HttpStatus.OK.value() == httpStatusCode.value()) {
```

Comment:
```suggestion
			if (HttpStatus.OK == httpStatusCode) {
```

You can can compare the enum directly.

Reply:
Thanks your advice. But:
`DefaultHttpStatusCode` and `HttpStatus`, they implement interface `HttpStatusCode`, use `#value` is correct. Otherwise a `DefaultHttpStatusCode` instance is NEVER `==` the `HttpStatus#OK`.

So I keep my code.